### PR TITLE
Add documentation for spack developer workflows

### PR DESCRIPTION
**Pull request by tmadlener**
Add some documentation on how to use `spack dev-build` and the development capabilities of spack environments for developer workflows. These capabilities of spack are not yet really documented, but can be quite useful.

- [x] `dev-build`
- [x] environments with `develop` packages

### CODE REVIEW

**Comment by tmadlener** [comment]
This should give a fairly comprehensive overview on the currently available spack development workflows. I haven't tested them very thoroughly in my daily work yet, but at least for the examples here, they do work surprisingly well.

File: spack-build-instructions/spack-developer-workflows.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
+# Spack workflows for developing Key4HEP software
+
+Using spack to develop software is somewhat pushing its intended usage to its limits.
+However, it is not impossible and this is an area of spack that is currently under active development.
+Unfortunately, this also means that the spack documentation might not be fully up-to-date on these topics.
```
Comment (vvolkl) [inline spack-build-instructions/spack-developer-workflows.md:5]: I think these sentences could be removed, as its clear that spack will be supporting this with all the dedicated commands.

File: spack-build-instructions/spack-developer-workflows.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+Hence, this page tries to collect some of the experiences the Key4HEP developers have made.
+
+## Developing a single package
+
+When only developing on a single package it is possible to use the [`dev-build`](https://spack.readthedocs.io/en/latest/command_index.html#spack-dev-build) command of spack.
+A brief tutorial can be found in [the spack documentation](https://spack-tutorial.readthedocs.io/en/lanl19/tutorial_developer_workflows.html).
```
Comment (vvolkl) [inline spack-build-instructions/spack-developer-workflows.md:11]:
```suggestion
A brief tutorial can be found in [the spack documentation](https://spack-tutorial.readthedocs.io/en/lanl19/tutorial_developer_workflows.html).
There is also a dedicated channel on slack (https://spackpm.herokuapp.com/) where questions regarding the development workflow can be discussed. 
```

**Review by vvolkl — APPROVED** [review]
Thanks a lot, this is a very nice and comprehensive write-up. I'd be  quite happy to have a CI job that somehow extracts the commands from these types of tutorials and runs them to check if anything breaks, I'll give it some thought how to best do this.

**Pull request merged**